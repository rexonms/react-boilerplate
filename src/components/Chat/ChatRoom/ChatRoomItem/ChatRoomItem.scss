.container {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
}
.message {
  padding: 10px;
  border-radius: 20px;
  p {
    margin: 0;
    font-size: $fontSizeSmall;
  }
}
.thumbnail {
  margin-right: 5px;
  div {
    width: 30px;
    height: 30px;
  }
}
.user {
  background: #ccc;
}
.friend {
  background: #eee;
}