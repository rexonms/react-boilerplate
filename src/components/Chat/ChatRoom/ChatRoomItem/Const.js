export const CHAT_BOX_FRIEND = 'friend';
export const CHAT_BOX_USER = 'user';
