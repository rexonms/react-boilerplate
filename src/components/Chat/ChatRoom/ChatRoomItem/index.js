import React from 'react';
import PropTypes from 'prop-types';

import Styles from './ChatRoomItem.scss';
import { CHAT_BOX_FRIEND, CHAT_BOX_USER } from './Const';
import Avatar from '../../Avatar/index';

const propTypes = {
  /** The chat message that needs to be dispalyed */
  message: PropTypes.string.isRequired,
  /** The chat message from the CHAT_BOX_FRIEND or CHAT_BOX_USER */
  type: PropTypes.oneOf([CHAT_BOX_FRIEND, CHAT_BOX_USER]).isRequired,
  /** Recipient thumbnailURl */
  imageURL: PropTypes.string,
  /** Recipient AltText */
  imageAltText: PropTypes.string,
};
const defaultProps = {
  imageURL: '',
  imageAltText: '',
};

const ChatRoomItem = props => (
  <div className={`${Styles.container} ${Styles[props.type]}`}>
    {props.type === CHAT_BOX_FRIEND &&
      <div className={Styles.thumbnail}>
        <Avatar imageURL={props.imageURL} imageAltText={props.imageAltText} />
      </div>
    }
    <div className={`side ${Styles.message}`}>
      <p>{props.message}</p>
    </div>
  </div>
);

ChatRoomItem.defaultProps = defaultProps;
ChatRoomItem.propTypes = propTypes;

export default ChatRoomItem;
