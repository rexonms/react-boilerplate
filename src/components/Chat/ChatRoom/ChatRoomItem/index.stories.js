import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import ChatRoomItem from './index';
import { CHAT_BOX_FRIEND, CHAT_BOX_USER } from './Const';

const stories = storiesOf('components/Chat/ChatRoom/ChatRoomItem', module);
const styles = {
  width: '500px',
  background: '#fff',
  border: '1px solid #ccc',
  padding: '20px',
};

export const props = {
  chatBoxTypes: [CHAT_BOX_FRIEND, CHAT_BOX_USER],
  message: 'How you doing?',
  imageURL: 'https://place-hold.it/100',
  imageAltText: 'Alt text of image',
}

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('ChatRoomItem', withInfo(`
  Component
  ~~~js
    <ChatRoomItem {...props}  />)
  ~~~
`)(() => {
  const type = select('type', props.chatBoxTypes, props.chatBoxTypes[0]);
  return (
    <div style={styles}> <ChatRoomItem {...props} type={type} /> </div>
  );
}));
