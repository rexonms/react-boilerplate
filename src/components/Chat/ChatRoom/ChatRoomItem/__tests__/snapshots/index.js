import React from 'react';
import { shallow } from 'enzyme';

import ChatRoomItem from '../../index';
import { chatBoxMessage, chatBoxTypes } from '../../index.stories';

// https://hackernoon.com/oh-snap-snapshots-unit-integration-testing-with-jest-enzyme-13cc18aecb7b
it('Renders component correctly', () => {
  const user = shallow(<ChatRoomItem message={chatBoxMessage} type={chatBoxTypes[0]} />);
  expect(user).toMatchSnapshot();
  const friend = shallow(<ChatRoomItem message={chatBoxMessage} type={chatBoxTypes[1]} />);
  expect(friend).toMatchSnapshot();
});
