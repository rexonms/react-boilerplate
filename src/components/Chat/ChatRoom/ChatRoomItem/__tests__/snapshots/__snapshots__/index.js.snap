// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatRoomItem
    message="\\"How you doing"
    type="friend"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        "How you doing
</p>,
      "className": "container friend",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "\\"How you doing",
      },
      "ref": null,
      "rendered": "\\"How you doing",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          "How you doing
</p>,
        "className": "container friend",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "\\"How you doing",
        },
        "ref": null,
        "rendered": "\\"How you doing",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Renders component correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatRoomItem
    message="\\"How you doing"
    type="user"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        "How you doing
</p>,
      "className": "container user",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "\\"How you doing",
      },
      "ref": null,
      "rendered": "\\"How you doing",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          "How you doing
</p>,
        "className": "container user",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "\\"How you doing",
        },
        "ref": null,
        "rendered": "\\"How you doing",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
